src/UCTSearch.cpp

```diff
@@ -71,11 +71,23 @@ class OutputAnalysisData {
 };
 
 
-UCTSearch::UCTSearch(GameState& g)
+UCTSearch::UCTSearch(GameState& g, const std::string & weightsfile)
```

Comment:
Meh. I don't think UCTSearch should know about a string indicating a file for the network. Wouldn't it make more sense if it took a Network& instead? GTP:: should then inject that ref into search.

This avoids some ugliness like GameState::place_free_handicap taking an UCTSearch*, even though it doesn't want an existing UCTSearch at all, it just really needs the info to set up the network for it.

You can probably also avoid that "global" search object you have in GTP now. The network that is specified on the commandline can be loaded an injected from the main function and cmdline parsing then.

Reply:
Yeah I agree.  Trying to remove the global Network reference is causing more trouble than necessary, making other places looking odd for no apparent benefit.  I'll move the network variable to GTP and just pass a reference.